doc/neps/nep-0031-uarray.rst

```diff
@@ -69,12 +69,27 @@ _will be discussed on the mailing list._
 
 The way we propose the overrides will be used by end users is::
 
-    from numpy import unumpy
-    TODO
+    from numpy import unumpy as np
```

Comment:
As soon as you do this, you're shadowing the main namespace which was imported before as `import numpy as np`. This has two issues:
(1) it's going through the `uarray` machinery for everything now, so not opt-in
(2) there's lots of junk in the main `numpy` namespace that we don't want to add to `unumpy`, so this doesn't even work if you use some obscure function further down (say `np.rate`).

I think to keep it optional, it should be 
```
from numpy import unumpy
with unumpy.set_backend(backend):   # this could be `np.set_backend`
    x = np.asarray(x)
```

Reply:
But the line `x = np.asarray` itself goes through NumPy and not `unumpy`, and therefore, `unumpy` isn't optional anymore.

Reply:
Or, flipping the argument, everything would need to be a multimethod.

Reply:
Or are you suggesting we monkey-patch everything on import of `unumpy`? Which is highly dubious, as if one library imports it, it's not optional at all.

Reply:
We just discussed this, but for the record - that was my muscle memory making a dumb typo, I meant:
```
from numpy import unumpy
with unumpy.set_backend(backend):   # this could be `np.set_backend`
    x = unumpy.asarray(x)
```